FilmWiki/FilmWiki/Controllers/FilmInfo/InfoFilmViewModel.swift

```diff
@@ -0,0 +1,31 @@
+//
+//  InfoFilmViewModel.swift
+//  FIlmWiki
+//
+//  Created by Мария on 19/04/2019.
+//  Copyright © 2019 vlad. All rights reserved.
+//
+
+import Foundation
+
+class InfoFilmViewModel<Service> where Service: NetworkService {
```

Comment:
Объясни мне необходимость этого шаблона здесь иначе удали

Reply:
У меня все сервисы наследуют протокол NetworkService. Во viewmodel я создаю объект этого сервиса, но хочу, чтобы он был типа NetworkService. Но так как в этом протоколе есть associatedtype, xcode мне не дает этого сделать: "Protocol 'NetworkService' can only be used as a generic constraint because it has Self or associated type requirements". Ну я и попытался выкрутиться

Reply:
Ага, а зачем тебе вьюмодель шаблонного типа? Ты собираешь ее переиспользовать? Можешь быть infoFilmViewModel может быть инициализирована с другим сервисом? Она будет иметь смысл?

Reply:
Хмм, наверное, в данном случае это излишне